packages/pwa-kit-dev/src/utils/script-utils.ts

```diff
@@ -84,6 +85,28 @@ export const getProjectPkg = async (): Promise<Pkg> => {
     }
 }
 
+/**
+ * Get the set of file paths within a specific directory
+ * @param dir Directory to walk
+ * @returns Set of file paths within the directory
+ */
+export const walkDir = async (dir: string, baseDir: string): Promise<Set<string>> => {
+    const fileSet: Set<string> = new Set()
+    const entries: Dirent[] = await readdir(dir, {withFileTypes: true})
+
+    for (const entry of entries) {
```

Comment:
How long does `walkDir` take? Particularly, how long does it take in large customer projects? With a `for...of` loop, it reads every directory sequentially. If you change to `await Promise.all(entries.map(...))`, then it could read directories in parallel, which may be faster.

Reply:
When reading `@salesforce/retail-react-app`, `walkDir` takes 11.195ms